### PR TITLE
feat(protocol): Ignore empty client hints

### PR DESCRIPTION
**Pull request by TBS1996**
if a client hint has an empty string, it should be treated as missing. 

#skip-changelog

### CODE REVIEW

**Review by olksdr — COMMENTED** [review]
Could you, please, also add unit tests to check this behaviour, if there are none?

**Comment by TBS1996** [comment]
> Could you, please, also add unit tests to check this behaviour, if there are none?

good call, newest commit has unit tests

File: relay-general/src/protocol/contexts/browser.rs [hunk 1]
```diff
@@ -152,6 +156,27 @@ mod tests {
...
+                        .into(),
+                ),
+            ))];
+            PairList(headers)
+        });
+
+        let client_hints = RawUserAgentInfo::from_headers(&headers).client_hints;
+
+        let from_hints = BrowserContext::from_client_hints(&client_hints);
+        assert!(from_hints.is_none())
```
Comment (iker-barriocanal) [inline relay-general/src/protocol/contexts/browser.rs]: nit: I'd remove these lines, and same in the files below
```suggestion
        let client_hints = RawUserAgentInfo::from_headers(&headers).client_hints;
        let from_hints = BrowserContext::from_client_hints(&client_hints);
        assert!(from_hints.is_none())
```

**Review by iker-barriocanal — APPROVED** [review]
Please add this PR to the changelog.

File: relay-general/src/protocol/contexts/browser.rs [hunk 2]
```diff
@@ -152,6 +156,27 @@ mod tests {
...
     }
 
+    #[test]
+    fn test_ignore_empty_browser() {
+        let headers = Headers({
+            let headers = vec![Annotated::new((
+                Annotated::new("SEC-CH-UA".to_string().into()),
+                Annotated::new(
+                    // browser field missing
+                    r#"Not_A Brand";v="99", "";v="109", "Chromium";v="109"#
```
Comment (olksdr) [inline relay-general/src/protocol/contexts/browser.rs]: What if the string will be 
```suggestion
                    r#"Not_A Brand";v="99", " ";v="109", "Chromium";v="109"#
```

in this case the tests won't work even though it is kind of empty. 

What you will have to do is to trim the string, when you check  for emptiness

File: relay-general/src/protocol/contexts/device.rs [hunk 3]
```diff
@@ -175,6 +175,11 @@ impl DeviceContext {
 impl FromUserAgentInfo for DeviceContext {
     fn from_client_hints(client_hints: &ClientHints) -> Option<Self> {
         let device = client_hints.sec_ch_ua_model?.to_owned();
+
+        if device.is_empty() {
```
Comment (olksdr) [inline relay-general/src/protocol/contexts/device.rs]: same here, and in the following test, you will have to `trim` the string here.

**Review by olksdr — COMMENTED** [review]
This look great! Thanks for adding the tests. 

One more thing is, please, trim the strings when you check if those are empty and then we can merge it.

File: relay-general/src/protocol/contexts/browser.rs (hunk 2 above)
Comment (TBS1996) [reply relay-general/src/protocol/contexts/browser.rs]: ok, I have now added trim to all the checks for emptiness

**Review by olksdr — APPROVED** [review]
